### PR TITLE
add static deriveKey

### PR DESCRIPTION
**Pull request by mafintosh**
Will add docs / tests, but wanna validate the approach first.

This is so it's easier to validate a passphrase before writing the file

### CODE REVIEW

**Comment by emilbayes** [comment]
Can you replace the internal call also to use this method?

**Comment by mafintosh** [comment]
@emilbayes updated

**Comment by emilbayes** [comment]
Should we add a static compare method also?

**Comment by mafintosh** [comment]
What would the signature for that be?

File: index.js [hunk 1]
```diff
@@ -233,6 +231,31 @@ class PasswordEncryptedOverlay {
     })
   }
 
+  static deriveKey (passphrase, settings, cb) {
+    assert(settings)
+    assert(settings.opslimit)
+    assert(settings.memlimit)
```
Comment (emilbayes) [inline index.js]: These need messages

**Pull request merged**